client/src/main/java/org/eclipse/hono/client/Command.java

```diff
@@ -169,6 +173,29 @@ public boolean isValid() {
         return valid;
     }
 
+    /**
+     * Gets info about why the command is invalid.
+     *
+     * @return Info string.
+     * @throws IllegalStateException if this command is valid.
+     */
+    public String getInvalidCommandReason() {
+        if (isValid()) {
+            throw new IllegalStateException("command is valid");
+        }
+        final StringJoiner joiner = new StringJoiner(", ");
+        if (message.getSubject() == null) {
+            joiner.add("subject not set");
+        }
+        if (message.getCorrelationId() == null && message.getMessageId() == null) {
+            joiner.add("message/correlation id not set");
+        }
+        if (message.getReplyTo() != null && originalReplyToId == null) {
+            joiner.add("invalid replyTo '" + message.getReplyTo() + "'");
+        }
+        return joiner.toString();
```

Comment:
wouldn't it make sense to do this in the factory method(s) already?

Reply:
yes, I've updated the class accordingly.